### ISSUE TITLE
Configure new nightly Jenkins job for module and aggregate Scoverage reports

### ISSUE THREAD

**Issue by helena** [akka/akka#26445]

**Comment by helena** [comment]
Looked at SupervisorHierarchySpec issue related to 
#24630, not sure how we'd prefer to proceed, because 

the test passes 
* when running alone
* when running in the regular akka nightly

and only fails 
* in a few Jenkins jobs
* if you run ```sbt akka-actor-tests/test```

Thus, a temporary workaround to unblock this task, I've added the following SBT flag in Jenkins for this new Scoverage nightly job's config:
```-Dakka.test.names.exclude=SupervisorHierarchySpec``` until the flaky test issue can be fixed.

**Comment by helena** [comment]
Will delete the [temporary test branch, jenkins-scoverage-nightly](https://github.com/akka/akka/branches) when I switch the job to run against master.

**Comment by helena** [comment]
* Initial run with Scoverage on [akka-actor only](https://jenkins.akka.io:8498/view/Nightly%20Jobs/job/akka-nightly-scoverage/19/scoverage-report/)

**Comment by helena** [comment]
Instrumented test coverage against akka-actor
<img width="477" alt="screen shot 2019-02-27 at 7 35 47 pm" src="https://user-images.githubusercontent.com/406103/53539599-0d722c80-3ac7-11e9-9275-83583acbf838.png">

**Comment by helena** [comment]
@johanandren brought up coverage of Java, which, having not done any java for so long, I'd forgotten about. Scoverage will not cover java code and it is specifically tuned for the Scala language. While https://github.com/sbt/sbt-jacoco can cover both, [this is a great post on why not to use one java coverage for scala as well, but a specific tool for each](http://buransky.com/scala/the-best-code-coverage-for-scala).

**Comment by helena** [comment]
So we can create another ticket for java

project/plugins.sbt
```addSbtPlugin("com.github.sbt" % "sbt-jacoco" % "<version>")```

AkkaDisciplinePlugin, some appropriate config of:
```
jacocoReportSettings := JacocoReportSettings()
  .withThresholds(
    JacocoThresholds(
      instruction = 80,
      method = 100,
      branch = 100,
      complexity = 100,
      line = 90,
      clazz = 100)
  )
```

And add this SBT task to the coverage nightly or to its own, depending on reporting directory behavior 
```jacocoAggregate```

**Comment by patriknw** [comment]
What we are looking for is to include coverage of tests written in Java into the scoverage report. The actual code of implementing the javadsl is still (typically) in Scala. For example akka.stream.javadsl will currently probably show up as 0% coverage, while that is not true because we have java tests exercising that part.

**Comment by helena** [comment]
Understood, and Scoverage does not cover java. So let's discuss.

**Comment by helena** [comment]
Closing. Let's just stick with the local per-module usage for now which is done.

Slightly related issue brought up by someone on the team https://github.com/scoverage/sbt-scoverage/issues/278, closing without following up.
